src/tests/src/solver/modelParser/testModelParser.cpp

```diff
@@ -312,12 +312,86 @@ BOOST_AUTO_TEST_CASE(test_library_model_multiple_variables)
     Antares::Solver::ModelParser::Library libraryObj = parser.parse(library);
     BOOST_REQUIRE_EQUAL(libraryObj.models.size(), 1);
     BOOST_REQUIRE_EQUAL(libraryObj.models[0].variables.size(), 2);
-    BOOST_CHECK_EQUAL(libraryObj.models[0].variables[0].name, "var_name1");
-    BOOST_CHECK_EQUAL(libraryObj.models[0].variables[0].lower_bound, 0);
-    BOOST_CHECK_EQUAL(libraryObj.models[0].variables[0].upper_bound, 1);
-    BOOST_CHECK_EQUAL(libraryObj.models[0].variables[1].name, "var_name2");
-    BOOST_CHECK_EQUAL(libraryObj.models[0].variables[1].lower_bound, -1);
-    BOOST_CHECK_EQUAL(libraryObj.models[0].variables[1].upper_bound, 2);
+    BOOST_CHECK_EQUAL(libraryObj.models[0].variables[0].id, "var_name1");
+    BOOST_CHECK_EQUAL(libraryObj.models[0].variables[0].lower_bound, "0");
+    BOOST_CHECK_EQUAL(libraryObj.models[0].variables[0].upper_bound, "1");
+    BOOST_CHECK_EQUAL(libraryObj.models[0].variables[1].id, "var_name2");
+    BOOST_CHECK_EQUAL(libraryObj.models[0].variables[1].lower_bound, "-1");
+    BOOST_CHECK_EQUAL(libraryObj.models[0].variables[1].upper_bound, "2");
+}
+
+// variable bounds are strings expressions
+BOOST_AUTO_TEST_CASE(test_library_model_variables_bounds)
+{
+    Antares::Solver::ModelParser::Parser parser;
+    const auto library = R"(
+        library:
+            id: "lib_id"
+            description: "lib_description"
+            port-types: []
+            models:
+                - id: "model_id"
+                  description: "model_description"
+                  parameters: []
+                  variables:
+                      - id: "var_name"
+                        lower-bound: "near-zero"
+                        upper-bound: "pmax"
+                  ports: []
+                  port-field-definitions: []
+                  constraints: []
+                  objective: "objective"
+        )"s;
+    Antares::Solver::ModelParser::Library libraryObj = parser.parse(library);
+    BOOST_CHECK_EQUAL(libraryObj.models[0].variables[0].id, "var_name");
+    BOOST_CHECK_EQUAL(libraryObj.models[0].variables[0].lower_bound, "near-zero");
+    BOOST_CHECK_EQUAL(libraryObj.models[0].variables[0].upper_bound, "pmax");
+}
+
+// variable variable-type
+BOOST_AUTO_TEST_CASE(test_library_model_variables_type)
+{
+    Antares::Solver::ModelParser::Parser parser;
```

Comment:
As instanciation of the parser can be found in every test :
```c++
Antares::Solver::ModelParser::Parser parser;
```
we could define a **fixture** to be given to every test as argument